%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Multiplicative domain poster
% Created by Nathaniel Johnston
% August 2009
% http://www.nathanieljohnston.com/2009/08/latex-poster-template/
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\documentclass[final]{beamer}
\usepackage[scale=1.24]{beamerposter}
\usecolortheme[RGB={0,0,144}]{structure}
\usetheme{AnnArbor}\usecolortheme{beaver}
\usepackage{graphicx}  		% allows us to import images
\usepackage{tikz}
\usepackage{verbatim,xmpmulti,color,multicol,multirow}
\usepackage{amssymb,amsmath,latexsym}
%-----------------------------------------------------------
% Custom commands that I use frequently
%-----------------------------------------------------------

\newcommand{\bb}[1]{\mathbb{#1}}
\newcommand{\cl}[1]{\mathcal{#1}}
\newcommand{\fA}{\mathfrak{A}}
\newcommand{\fB}{\mathfrak{B}}
\newcommand{\Tr}{{\rm Tr}}
\newtheorem{thm}{Theorem}

%-----------------------------------------------------------
% Define the column width and poster size
% To set effective sepwid, onecolwid and twocolwid values, first choose how many columns you want and how much separation you want between columns
% The separation I chose is 0.024 and I want 4 columns
% Then set onecolwid to be (1-(4+1)*0.024)/4 = 0.22
% Set twocolwid to be 2*onecolwid + sepwid = 0.464
%-----------------------------------------------------------

\newlength{\sepwid}
\newlength{\onecolwid}
\newlength{\twocolwid}
\setlength{\paperwidth}{48in}
\setlength{\paperheight}{36in}
\setlength{\sepwid}{0.024\paperwidth}
\setlength{\onecolwid}{0.22\paperwidth}
\setlength{\twocolwid}{0.464\paperwidth}
\setlength{\topmargin}{-1in}
\usetheme{confposter}
\usepackage{exscale}

%-----------------------------------------------------------
% The next part fixes a problem with figure numbering. Thanks Nishan!
% When including a figure in your poster, be sure that the commands are typed in the following order:
% \begin{figure}
% \includegraphics[...]{...}
% \caption{...}
% \end{figure}
% That is, put the \caption after the \includegraphics
%-----------------------------------------------------------

\usecaptiontemplate{
\small
\structure{\insertcaptionname~\insertcaptionnumber:}
\insertcaption}

%-----------------------------------------------------------
% Define colours (see beamerthemeconfposter.sty to change these colour definitions)
%-----------------------------------------------------------

\setbeamercolor{block title}{fg=ngreen,bg=white}
\setbeamercolor{block body}{fg=black,bg=white}
\setbeamercolor{block alerted title}{fg=white,bg=dblue!70}
\setbeamercolor{block alerted body}{fg=black,bg=white!10}

%-----------------------------------------------------------
% Name and authors of poster/paper/research
%-----------------------------------------------------------

\title{Accounting for Nuisance Covariates when Using
RNA-Seq Data \\
to Identify Differentially Expressed Genes}
\author{Yet Nguyen and Dan Nettleton, Department of Statistics, Iowa State University}
\date{August 04, 2014}
%\institute{}

%-----------------------------------------------------------
% Start the poster itself
%-----------------------------------------------------------
% The \rmfamily command is used frequently throughout the poster to force a serif font to be used for the body text
% Serif font is better for small text, sans-serif font is better for headers (for readability reasons)
%-----------------------------------------------------------

\begin{document}
<<chunk_options, echo=FALSE, message=FALSE>>=
opts_chunk$set(fig.width=8, fig.height=5, fig.align='center', size='tiny')
####################################
#  #
####################################
library(reshape)
library(ggplot2)
library(dplyr)
library(gridExtra)
@

\addtobeamertemplate{headline}{} 
{\begin{tikzpicture}[remember picture, overlay]
     \node [anchor=north east, inner sep=3cm]  at (current page.north east)
     {\includegraphics[height=5cm]{isu.png}};
  \end{tikzpicture}}

\begin{frame}[t,fragile]
  \begin{columns}[t]												% the [t] option aligns the column's content at the top
    \begin{column}{\sepwid}\end{column}			% empty spacer column
    \begin{column}{\onecolwid}
      \begin{alertblock}{Introduction}
        \rmfamily{ Suppose that we have RNA sequencing (RNA-seq) read count data  
        on $J$ genes from an experiment with two treatment groups. We want to find genes that are differentially expressed (DE) 
        between two treatment groups. Analyses can often be complicated by the presence of nuisance 
        factors that arise due to experimental design limitations and 
        heterogeneity of experimental units that can be seen in 
        continuous covariates measured for each experimental unit 
        and/or RNA sample.}
      \end{alertblock}
%       \vskip2ex
%       \begin{block}{Prototypical Dataset}
%         \rmfamily{
%         \begin{table} [h!]
% \centering
% \renewcommand{\arraystretch}{1.5}
% \begin{tabular}{CCCCCCCCC}
% \toprule
% \multicolumn{1}{l}{ }                 &
% \multicolumn{4}{c}{Treatment 1}      &
% \multicolumn{4}{c}{Treatment 2}        \\ 
% % \cmidrule(lr){2-5}
% % \cmidrule(lr){6-9}
% &
% \multicolumn{1}{c}{$u_{11}$}           &
% \multicolumn{1}{c}{$u_{12}$}           &
% \multicolumn{1}{c}{$\cdots$}            &
% \multicolumn{1}{c}{$u_{1n}$}           &
% \multicolumn{1}{c}{$u_{21}$}           &
% \multicolumn{1}{c}{$u_{22}$}           &
% \multicolumn{1}{c}{$\cdots$}            &
% \multicolumn{1}{c}{$u_{2n}$}            \\
% \midrule
% x            & 0.5    & 0.95   & \dots  & -1.42    & -.45   & .89    & \dots  & 1.2  \\
% \nm{gene 1}  & 56     & 2014   & \cdots & 28       & 31     & 975    & \cdots & 3289 \\ 
% \nm{gene 2}  & 0      & 2      & \cdots & 1        & 0      & 0      & \cdots & 1 \\
% \nm{gene 3}  & 1      & 3      & \cdots & 0        & 0      & 0      & \cdots & 0 \\
% \vdots       & \vdots & \vdots &        & \vdots   & \vdots & \vdots &        & \vdots\\
% \nm{gene }J  & 1701   & 264    & \cdots & 345      & 14     & 234    & \cdots & 34 \\
% 
% %\midrule[\heavyrulewidth]
% \bottomrule
% \end{tabular}
% \end{table}
%         }
%       \end{block}
%       \vskip2ex
     \vskip2ex
       \begin{block}{Scenarios Considered}
            \rmfamily{
            <<echo =FALSE, warning=FALSE, fig.align='center', fig.show='hold'>>=  
par(mfrow = c(2,2))
# plot1
x <- c(-2, 2)
y <- c(-1, 1)
plot(x, y, ylab = "log(mean)",
          xlab = "x", xlim = x, ylim = y ,
     type = "n", 
     cex.lab=2, cex.axis=2, cex.main=2, cex.sub=2)
abline(h =.3, col = "purple")
# plot2
x <- c(-2, 2)
y <- c(-1, 1)
plot(x, y, ylab = "log(mean)",
     xlab = "x", xlim = x, ylim = y ,
     type = "n", 
     cex.lab=2, cex.axis=2, cex.main=2, cex.sub=2)
lines(c(-3,3), c(-0.5, 1), col = "purple")
# plot3
x <- c(-2, 2)
y <- c(-1, 1)
plot(x, y, ylab = "log(mean)",
     xlab = "x", xlim = x, ylim = y ,
     type = "n", 
     cex.lab=2, cex.axis=2, cex.main=2, cex.sub=2)
abline(h =.3, col = "blue")
abline(h =-.2, col = "red")
# plot4
x <- c(-2, 2)
y <- c(-1, 1)
plot(x, y, ylab = "log(mean)",
     xlab = "x", xlim = x, ylim = y ,
     type = "n", 
     cex.lab=2, cex.axis=2, cex.main=2, cex.sub=2)
lines(c(-3,3), c(-0.3, 1.2), col = "blue")

lines(c(-3,3), c(-0.6, 0.9), col = "red")
@
            }
\end{block}
\vskip2ex
 \begin{alertblock}{Analysis Strategies}
        \rmfamily{
        \begin{itemize} 
        \item Ignore the covariate.%, i.e., assume the covariate is not 
      %associated with the transcript abundance of any gene.
      
      \item Include the covariate in a model for each gene. %, i.e., 
      %assume the covariate may be associated with the transcript
      %abundance of each gene.
      \item Assume the covariate is associated with transcript abundance 
      of some subset of genes and consider model selection criteria 
      or model averaging strategies when performing inference
      for treatment effects.
        \end{itemize}
        }
      \end{alertblock}
      \vskip2ex
\begin{alertblock}{Model for Simulated Data}
\rmfamily{
$i =1, 2$;  $j = 1, \cdots, J$; $k = 1, \cdots, n$ 
 \begin{equation*}
y_{ijk} \sim \mbox{NegBin}(\mu_{ijk}, \omega_{j}),\quad 
 \log(\mu_{ijk}) = \tau_{ij} +\beta_j x_{k},
\end{equation*}
where
\begin{itemize}
 \item  $n = 5$ or $n = 20$ % experimential units per treatment
 \item $x_{k}\sim N(0,1).$
 \end{itemize}
}
\end{alertblock}

\vskip2ex
    \end{column}
    \begin{column}{\sepwid}\end{column}			% empty spacer column
    \begin{column}{\onecolwid}							% create a two-column-wide column and then we will split it up later
         

\begin{alertblock}{Parameters Used for Simulation}
\rmfamily{
\begin{itemize}
 %\setlength{\itemsep}{.2in}
 \item $\tau_{1j}-\tau_{2j}$ represents the log fold change for gene $j$.
 
 \item $\omega_j$ is the negative binomial dispersion parameter.
 
 \item Values for $\tau_{1j} - \tau_{2j}$ and $\omega_j$ were simulated
       to match values estimated from real data.
       
 \item In all simulations, 80\% of the $\tau_{1j}-\tau_{2j}$ values
       were set to zero among $J = 1000$ genes.
\item $\beta_j \sim \frac{\nu}{2}\times\mbox{Unif}(L,U)+ \frac{\nu}{2}\times(-\mbox{Unif}(L, U))\\$ $\qquad\quad + (1-\nu)\times \delta_{\{0\}} $
\item $\nu \in \{0, 0.25, 0.50, 0.75\}$
\item $(L,U) \in \{(0.1, 0.5),(1, 1.5)\} $
\end{itemize}
}
\end{alertblock}

\begin{alertblock}{Testing for Trt and Cov Effects}
\rmfamily{
\begin{itemize}
\item The \texttt{QuasiSeq} R package (Lund et al., 2012) was used to 
      obtain a $p$-value for each test.
\item $p_{j\tau}$ is the $p$-value for the test of $H_{0j}: \tau_{1j}=\tau_{2j} $
      for the no-covariate model, $\log(\mu_{ijk}) = \tau_{ij}$.

\item $p_{j\tau|\beta}$ is the $p$-value for the test of $H_{0j}: \tau_{1j}=\tau_{2j} $
      for the covariate model, $\log(\mu_{ijk}) = \tau_{ij} +\beta_j x_k$.

\item $p_{j\beta|\tau}$ is the $p$-value for the test of $H_{0j}: \beta_j = 0 $
      for the covariate model, $\log(\mu_{ijk}) = \tau_{ij} +\beta_j x_k$.
\end{itemize}     
}
\end{alertblock}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{alertblock}{Methods for Identification of DE}
\rmfamily{
% We consider 5 methods for converting $p_{j\tau}, p_{j\tau|\beta}, p_{j\beta|\tau}$
% to a decision about the DE status of gene $j$.
\begin{itemize}
%\setlength{\itemsep}{.2in}
\item [1.] nocov: Convert $p_{j\tau}\; (j = 1,\cdots, J)$ to $q$-values. 

\item [2.] cov: Convert $p_{j\tau|\beta} \;(j = 1,\cdots, J)$ to $q$-values.

\item [3.] ebp: Convert 
           \[
           p_{j\tau}I[\mbox{EBP}(p_{j\beta|\tau})>0.5] + 
           p_{j\tau|\beta}I[\mbox{EBP}(p_{j\beta|\tau})\leq 0.5] 
           \;
           \]
           $ (j = 1,\dots, J)$ to $q$-values. 

\item [4.] aic: Convert 
           \[
           p_{j\tau}I[\mbox{AIC}_{j\tau} <\mbox{AIC}_{j\tau\beta}]+
           p_{j\tau|\beta}I[\mbox{AIC}_{j\tau} \geq \mbox{AIC}_{j\tau\beta}] 
           \;
           \]
           $ (j = 1,\dots, J)$ to $q$-values. 
           
\item [5.]  aaa: Compute
           \begin{align*}
           \mbox{EBP}_j &= \mbox{EBP}(p_{j\tau})\mbox{EBP}(p_{j\beta|\tau})+\\
           &\mbox{EBP}(p_{j\tau|\beta})[1-\mbox{EBP}(p_{j\beta|\tau})]
           \end{align*}
           $ (j = 1,\dots, J)$.  (Pounds and Rai (2009), Pounds et al. (2012)).
\end{itemize}    
}
\end{alertblock}
\vskip2ex
\end{column}
%%%%%%%%%%%%%%


    \begin{column}{\sepwid}\end{column}  		% empty spacer column
    \begin{column}{\twocolwid}							% create a two-column-wide column and then we will split it up later
      \begin{columns}[t,totalwidth=\twocolwid]	% split up that two-column-wide column
        \begin{column}{\onecolwid}\vspace{-.69in}
          \begin{alertblock}{Computing EBP}
\rmfamily{
\begin{itemize}
%\setlength{\itemsep}{.2in}

\item Suppose $p_j$ is a $p$-value for testing a null 
      hypothesis $H_{0j}$ for $j = 1, \cdots, J$.

\item 
      $
      \mbox{EBP}(p_j):= \widehat{\mbox{P}}(H_{0j}|p_j) = 
      \frac{\hat{\pi}_0}{\hat{f}(p_j)},
      $
      where $\hat{f}$ is the Grenander estimator of the pdf of $p$-values, 
      which is the nonparametric MLE of  decreasing pdf of $p$-values, and
      $\hat{\pi}_0 = \hat{f}(1)$. 
      
      (Efron et al. (2001), Strimmer (2008), Grenander (1956)).
      
\end{itemize}    
}
\end{alertblock}

        \end{column}
        \begin{column}{\onecolwid}\vspace{-.69in}
              \begin{alertblock}{Evaluation of Methods}
      \rmfamily{
      \begin{itemize}
 \item For  100 replications of each simulation setting, we compute
 \begin{itemize}
 
 \item Average Area Under the Receiver Operating Characteristic Curve
 
 \item Estimated FDR when FDR is nominally controlled at 0.05
 \end{itemize}
 
 \item We use Storey (2002)'s FDR control procedure based on $q$-values for Method 1, 2, 3, and 4, and average accummulative EBP for Method 5.

\end{itemize}
      }
    \end{alertblock}
        \end{column}
      \end{columns}
      \begin{block}{Simulation Results}		% an ACTUAL two-column-wide column
        \rmfamily{
        <<aucall_05, echo = FALSE, results = 'asis',fig.show='hold',warning=FALSE,message=FALSE>>=
wd <- "/home/ntyet/research"
load(file = paste(wd, "/K_20/pbeta_0/out_20_0.RData", sep = ""))
load(file = paste(wd, "/K_20/pbeta_0.25/out_20_25.RData", sep = ""))
load(file = paste(wd, "/K_20/pbeta_0.5/out_20_50.RData", sep = ""))
load(file = paste(wd, "/K_20/pbeta_0.75/out_20_75.RData", sep = ""))
load(file = paste(wd, "/K_5/pbeta_0/out_5_0.RData", sep = ""))
load(file = paste(wd, "/K_5/pbeta_0.25/out_5_25.RData", sep = ""))
load(file = paste(wd, "/K_5/pbeta_0.5/out_5_50.RData", sep = ""))
load(file = paste(wd, "/K_5/pbeta_0.75/out_5_75.RData", sep = ""))
K_5_beta_1_auc_out2 <- as.data.frame(rbind(out_5_0[[1]][,8+c(1,2,4,5,7,8)], 
                                           out_5_25[[1]][,8+c(1,2,4,5,7,8)], 
                                           out_5_50[[1]][,8+c(1,2,4,5,7,8)], 
                                           out_5_75[[1]][,8+c(1,2,4,5,7,8)]))


K_5_beta_2_auc_out2 <- as.data.frame(rbind(out_5_0[[2]][,8+c(1,2,4,5,7,8)], 
                                           out_5_25[[2]][,8+c(1,2,4,5,7,8)], 
                                           out_5_50[[2]][,8+c(1,2,4,5,7,8)], 
                                           out_5_75[[2]][,8+c(1,2,4,5,7,8)]))

colnames(K_5_beta_1_auc_out2) <- 
  colnames(K_5_beta_2_auc_out2) <- 
  c("nocov", "cov", "ebp", "aic", "aaa", "oracle")
K_5_beta_1_auc_out2$nu <- K_5_beta_2_auc_out2$nu <- rep(as.character(c(0,25,50,75)), each = 100)
K_5_auc_out2 <- rbind(K_5_beta_1_auc_out2, K_5_beta_2_auc_out2)

K_5_auc_out2$setup <- rep(c("(L,U)=(0.1,0.5)", "(L,U)=(1,1.5)"), 
                         each = dim(K_5_beta_1_auc_out2)[1])


K_5_auc_out2_melt <- melt(K_5_auc_out2)
colnames(K_5_auc_out2_melt)[3:4] <- c("Method", "PAUROC") 

K_5_auc_out2_mean <- 
K_5_auc_out2_melt %.%
  group_by(nu, setup, Method) %.%
  summarise(PAUROC = mean(PAUROC))

K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "nocov",][,4] <- 
  mean(K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "nocov",][,4] )

K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "cov",][,4] <- 
  mean(K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "cov",][,4] )

K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "ebp",][,4] <- 
  mean(K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "ebp",][,4] )

K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "aic",][,4] <- 
  mean(K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "aic",][,4] )

K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "aaa",][,4] <- 
  mean(K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "aaa",][,4] )

K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "oracle",][,4] <- 
  mean(K_5_auc_out2_mean[K_5_auc_out2_mean$nu==0 & K_5_auc_out2_mean$Method == "oracle",][,4] )

#save(K_5_auc_out2_mean, file = paste(wd, "/K_5/K_5_auc_out2_mean.RData", sep = ""))

#load(file = paste(wd, "/K_5/K_5_auc_out2_mean.RData", sep = ""))
p <- ggplot(K_5_auc_out2_mean, aes(nu, PAUROC, colour = Method, group = Method)) + geom_line()
p1 <- p + facet_grid(.~setup )+
  xlab(expression(paste(nu)))+
  theme(axis.text=element_text(size=20),
        axis.title=element_text(size=20))+ 
      theme(strip.text.x = element_text(size=20))+
  theme(legend.text = element_text(size=20))+ 
  theme(legend.title = element_text(size=20))+
  ggtitle("PAUROC (FPR < 0.05), n = 5")+ 
     theme(plot.title = element_text(size=20,lineheight=.8, face="bold"))

@



<<fdpall, echo = FALSE, results = 'asis',fig.show='hold',warning=FALSE,message=FALSE>>=

K_5_beta_1_fdp_out <- as.data.frame(rbind(out_5_0[[1]][,16+c(1,2,4,5,7,8)], 
                                           out_5_25[[1]][,16+c(1,2,4,5,7,8)], 
                                           out_5_50[[1]][,16+c(1,2,4,5,7,8)], 
                                           out_5_75[[1]][,16+c(1,2,4,5,7,8)]))


K_5_beta_2_fdp_out <- as.data.frame(rbind(out_5_0[[2]][,16+c(1,2,4,5,7,8)], 
                                           out_5_25[[2]][,16+c(1,2,4,5,7,8)], 
                                           out_5_50[[2]][,16+c(1,2,4,5,7,8)], 
                                           out_5_75[[2]][,16+c(1,2,4,5,7,8)]))

colnames(K_5_beta_1_fdp_out) <- 
  colnames(K_5_beta_2_fdp_out) <- 
  c("nocov", "cov", "ebp", "aic", "aaa", "oracle")
K_5_beta_1_fdp_out$nu <- K_5_beta_2_fdp_out$nu <- rep(as.character(c(0,25,50,75)), each = 100)
K_5_fdp_out <- rbind(K_5_beta_1_fdp_out, K_5_beta_2_fdp_out)

K_5_fdp_out$setup <- rep(c("(L,U)=(0.1,0.5)", "(L,U)=(1,1.5)"), 
                         each = dim(K_5_beta_1_fdp_out)[1])


K_5_fdp_out_melt <- melt(K_5_fdp_out)
colnames(K_5_fdp_out_melt)[3:4] <- c("Method", "FDR") 

K_5_fdp_out_mean <- 
K_5_fdp_out_melt %.%
  group_by(nu, setup, Method) %.%
  summarise(FDR = mean(FDR))

K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "nocov",][,4] <- 
  mean(K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "nocov",][,4] )

K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "cov",][,4] <- 
  mean(K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "cov",][,4] )

K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "ebp",][,4] <- 
  mean(K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "ebp",][,4] )

K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "aic",][,4] <- 
  mean(K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "aic",][,4] )

K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "aaa",][,4] <- 
  mean(K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "aaa",][,4] )

K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "oracle",][,4] <- 
  mean(K_5_fdp_out_mean[K_5_fdp_out_mean$nu==0 & K_5_fdp_out_mean$Method == "oracle",][,4] )

#save(K_5_fdp_out_mean, file = paste(wd, "/K_5/K_5_fdp_out_mean.RData", sep = ""))

#load(file = paste(wd, "/K_5/K_5_fdp_out_mean.RData", sep = ""))

p <- ggplot(K_5_fdp_out_mean, aes(nu, FDR, colour = Method, group = Method)) + geom_line()
p2 <- p + facet_grid(.~setup )+
  xlab(expression(paste(nu)))+
  theme(axis.text=element_text(size=20),
        axis.title=element_text(size=20))+ 
      theme(strip.text.x = element_text(size=20))+
  theme(legend.text = element_text(size=20))+ 
  theme(legend.title = element_text(size=20))+
  ylab("Estimated FDR")+theme(axis.text=element_text(size=20),
        axis.title=element_text(size=20))+
  geom_hline(yintercept = 0.05)+
  ggtitle("Control FDR at 5%, n = 5")+ 
     theme(plot.title = element_text(size=20,lineheight=.8, face="bold"))

@



<<paucall_20, echo = FALSE, results = 'asis',fig.show='hold',warning=FALSE,message=FALSE>>=

K_20_beta_1_auc_out2 <- as.data.frame(rbind(out_20_0[[1]][,8+c(1,2,4,5,7,8)], 
                                           out_20_25[[1]][,8+c(1,2,4,5,7,8)], 
                                           out_20_50[[1]][,8+c(1,2,4,5,7,8)], 
                                           out_20_75[[1]][,8+c(1,2,4,5,7,8)]))


K_20_beta_2_auc_out2 <- as.data.frame(rbind(out_20_0[[2]][,8+c(1,2,4,5,7,8)], 
                                           out_20_25[[2]][,8+c(1,2,4,5,7,8)], 
                                           out_20_50[[2]][,8+c(1,2,4,5,7,8)], 
                                           out_20_75[[2]][,8+c(1,2,4,5,7,8)]))

colnames(K_20_beta_1_auc_out2) <- 
  colnames(K_20_beta_2_auc_out2) <- 
  c("nocov", "cov", "ebp", "aic", "aaa", "oracle")
K_20_beta_1_auc_out2$nu <- K_20_beta_2_auc_out2$nu <- rep(as.character(c(0,25,50,75)), each = 100)
K_20_auc_out2 <- rbind(K_20_beta_1_auc_out2, K_20_beta_2_auc_out2)

K_20_auc_out2$setup <- rep(c("(L,U)=(0.1,0.5)", "(L,U)=(1,1.5)"), 
                         each = dim(K_20_beta_1_auc_out2)[1])


K_20_auc_out2_melt <- melt(K_20_auc_out2)
colnames(K_20_auc_out2_melt)[3:4] <- c("Method", "PAUROC") 

#colnames(K_20_auc_out2_melt)

K_20_auc_out2_mean <- 
K_20_auc_out2_melt %.%
  group_by(nu, setup, Method) %.%
  summarise(PAUROC = mean(PAUROC))
K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "nocov",][,4] <- 
  mean(K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "nocov",][,4] )

K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "cov",][,4] <- 
  mean(K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "cov",][,4] )

K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "ebp",][,4] <- 
  mean(K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "ebp",][,4] )

K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "aic",][,4] <- 
  mean(K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "aic",][,4] )

K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "aaa",][,4] <- 
  mean(K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "aaa",][,4] )

K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "oracle",][,4] <- 
  mean(K_20_auc_out2_mean[K_20_auc_out2_mean$nu==0 & K_20_auc_out2_mean$Method == "oracle",][,4] )
#save(K_20_auc_out2_mean, file = paste(wd, "/K_20/K_20_auc_out2_mean.RData", sep = ""))

#load(file = paste(wd, "/K_20/K_20_auc_out2_mean.RData", sep = ""))
p <- ggplot(K_20_auc_out2_mean, aes(nu, PAUROC, colour = Method, group = Method)) + geom_line()
p3 <- p + facet_grid(.~setup )+
  xlab(expression(paste(nu)))+
  theme(axis.text=element_text(size=20),
        axis.title=element_text(size=20))+ 
      theme(strip.text.x = element_text(size=20))+
  theme(legend.text = element_text(size=20))+ 
  theme(legend.title = element_text(size=20))+
  ggtitle("PAUROC (FPR < 0.05), n = 20")+ 
     theme(plot.title = element_text(size=20,lineheight=.8, face="bold"))


@





<<fdpall20, echo = FALSE, results = 'asis',fig.show='hold',warning=FALSE,message=FALSE>>=

K_20_beta_1_fdp_out <- as.data.frame(rbind(out_20_0[[1]][,16+c(1,2,4,5,7,8)], 
                                           out_20_25[[1]][,16+c(1,2,4,5,7,8)], 
                                           out_20_50[[1]][,16+c(1,2,4,5,7,8)], 
                                           out_20_75[[1]][,16+c(1,2,4,5,7,8)]))


K_20_beta_2_fdp_out <- as.data.frame(rbind(out_20_0[[2]][,16+c(1,2,4,5,7,8)], 
                                           out_20_25[[2]][,16+c(1,2,4,5,7,8)], 
                                           out_20_50[[2]][,16+c(1,2,4,5,7,8)], 
                                           out_20_75[[2]][,16+c(1,2,4,5,7,8)]))

colnames(K_20_beta_1_fdp_out) <- 
  colnames(K_20_beta_2_fdp_out) <- 
  c("nocov", "cov", "ebp", "aic", "aaa", "oracle")
K_20_beta_1_fdp_out$nu <- K_20_beta_2_fdp_out$nu <- rep(as.character(c(0,25,50,75)), each = 100)
K_20_fdp_out <- rbind(K_20_beta_1_fdp_out, K_20_beta_2_fdp_out)

K_20_fdp_out$setup <- rep(c("(L,U)=(0.1,0.5)", "(L,U)=(1,1.5)"), 
                         each = dim(K_20_beta_1_fdp_out)[1])


K_20_fdp_out_melt <- melt(K_20_fdp_out)
colnames(K_20_fdp_out_melt)[3:4] <- c("Method", "FDR") 

#colnames(K_20_fdp_out_melt)

K_20_fdp_out_mean <- 
K_20_fdp_out_melt %.%
  group_by(nu, setup, Method) %.%
  summarise(FDR = mean(FDR))

K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "nocov",][,4] <- 
  mean(K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "nocov",][,4] )

K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "cov",][,4] <- 
  mean(K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "cov",][,4] )

K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "ebp",][,4] <- 
  mean(K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "ebp",][,4] )

K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "aic",][,4] <- 
  mean(K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "aic",][,4] )

K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "aaa",][,4] <- 
  mean(K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "aaa",][,4] )

K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "oracle",][,4] <- 
  mean(K_20_fdp_out_mean[K_20_fdp_out_mean$nu==0 & K_20_fdp_out_mean$Method == "oracle",][,4] )
#save(K_20_fdp_out_mean, file = paste(wd, "/K_20/K_20_fdp_out_mean.RData", sep = ""))

#load(file = paste(wd, "/K_20/K_20_fdp_out_mean.RData", sep = ""))

p <- ggplot(K_20_fdp_out_mean, aes(nu, FDR, colour = Method, group = Method)) + geom_line()

p4 <- p + facet_grid(.~setup )+
  xlab(expression(paste(nu)))+
  theme(axis.text=element_text(size=20),
        axis.title=element_text(size=20))+ 
      theme(strip.text.x = element_text(size = 20))+
  theme(legend.text = element_text(size = 20))+ 
  theme(legend.title = element_text(size = 20))+
  ylab("Estimated FDR")+theme(axis.text=element_text(size=20),
        axis.title=element_text(size=20))+
  geom_hline(yintercept = 0.05)+
  ggtitle("Control FDR at 5%, n = 20")+ 
     theme(plot.title = element_text(size=20,lineheight=.8, face="bold"))

@

<<echo=FALSE,warning=FALSE,message=FALSE,error=FALSE,fig.width=16,fig.height=7>>=
grid.arrange( p1, p2, p3,p4, ncol=2)

@
        
        }
      \end{block}
      
      \begin{alertblock}{Comments on the Results and Conclusions}
          \rmfamily{
\begin{itemize}
\item For the simulation settings we considered, the cov method based on 
      $p_{j\tau|\beta}$  performed best overall.
      
\item The price paid for including an irrelevant covariate in the model 
      was far less than the cost of excluding an important covariate. 
      
\item All other methods relied on $p_{j\tau}$  to some 
      extent and performed poorly whenever a substantial portion of genes 
      were associated with the covariate.
      \item When $\beta_j \neq 0,$ $p_{j\tau}$ is likely to be inflated because 
      variation in the response unexplained by the model pushes the test 
      of $H_{0j}: \tau_{1j} = \tau_{2j}$ toward nonsignificance.

\item Even when $\beta_j =0$, $p_{j\tau}$ is likely to be inflated if 
      $\beta_{j^*} \neq 0$ for a substantial portion of genes $j^* \neq j$.
      
\item Borrowing information across genes to estimate gene-specific dispersions
      can lead to overestimation of dispersions when $\beta_{j^*}\neq 0$ 
      for many genes and the covariate is excluded from the model for each gene. 
\item When using existing software for RNA-seq analysis that requires the model 
      for each gene to have the same design matrix, it may be best to favor flexibility
      over simplicity.

\item More sophisticated empirical Bayes of fully Bayesian strategies for 
      combining model selection and inference are needed.
\end{itemize}
          }
        \end{alertblock}
        
      \begin{alertblock}{Acknowledgements}  
      This material is based upon work supported by Agriculture and Food
Research Initiative Competitive Grant No. 2011-68004-30336 from
the USDA National Institute of Food and Agriculture, and the National Science Foundation 
(NSF) under Grant No. 0922746.

      \end{alertblock}
      
  \end{column}
  \begin{column}{\sepwid}\end{column}			% empty spacer column
 \end{columns}
\end{frame}
\end{document}
